src/components/Sidebar/CommentList/CommentItem.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useContext } from 'react';
+import { Comment } from '../../../types/Comment';
+import { CommentListContext } from '../../../utils/Store';
+import { deleteComment } from '../../../utils/fetchFunctions';
+
+interface CommentItemProps {
+  comment: Comment;
+}
+
+export const CommentItem: React.FC<CommentItemProps> = ({ comment }) => {
+  const { comments, setComments } = useContext(CommentListContext);
+
+  function handleDeleteButton() {
+    const newList = comments.filter(item => item.id !== comment.id);
+    const copy = [...comments];
+
+    setComments(newList);
+
+    deleteComment(comment.id).catch(() => {
+      alert('Failed to delete comment, please try again!');
+
+      setComments(copy);
+    });
```

Comment:
It's recommended to use 'try {..} catch' for error handling instead of '.catch()' for better readability and consistency in handling asynchronous operations.